File: hservice/src/main/java/com/hservice/models/Task.java
package com.hservice.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.Objects;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long taskId;
    private String taskName;
    private String taskCode;
    private long project;
    private long taskCreator;
    private long taskExecutor;
    private Timestamp dueDate;

    @CreationTimestamp
    private Timestamp created;

    @UpdateTimestamp
    private Timestamp updated;

    @ManyToOne(fetch = FetchType.EAGER,cascade = CascadeType.MERGE)
    @JoinColumn(name = "type", nullable = false)
    private Type type;

    @ManyToOne
    @JoinColumn(name = "priority", nullable = false)
    private Priority priority;

    @ManyToOne
    @JoinColumn(name = "status",  nullable = false)
    private Status status;

    @ManyToOne
    @JoinColumn(name = "description")
    private Description description;
}

File: frontend/src/app/dto/models/task.ts
import {Type} from "./type";
import {Priority} from "./priority";
import {Status} from "./status";
import {Description} from "./description";

export class Task {
  taskId: number;
  taskName: string;
  taskCode: string;
  dueDate: Date;
  created: Date;
  updated: Date;
  type: Type;
  priority: Priority;
  status: Status;
  description: Description;
  project: number;
  taskCreator: number;
  taskExecutor: number;
}
